drivers/syslog/syslog_rpmsg_server.c

```diff
@@ -200,11 +283,16 @@ static int syslog_rpmsg_ept_cb(FAR struct rpmsg_endpoint *ept,
           memcpy(priv->tmpbuf + priv->nextpos, msg->data + printed, copied);
           priv->nextpos += copied;
         }
+    }
+#ifdef CONFIG_SYSLOG_RPMSG_SERVER_CHARDEV
+  else if (header->command == SYSLOG_RPMSG_SYNC)
+    {
+      FAR struct syslog_rpmsg_sync_s *msg = data;
+      sem_t *sem = (FAR sem_t *)(uintptr_t)msg->cookie;
```

Comment:
```suggestion
      FAR sem_t *sem = (FAR sem_t *)(uintptr_t)msg->cookie;
```

Reply:
Done. update in https://github.com/apache/incubator-nuttx/pull/6655. Thanks.